src/test/java/com/commercetools/sync/taxcategories/TaxCategorySyncOptionsTest.java

```diff
@@ -0,0 +1,158 @@
+package com.commercetools.sync.taxcategories;
+
+import com.commercetools.sync.commons.utils.TriFunction;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.taxcategories.TaxCategory;
+import io.sphere.sdk.taxcategories.TaxCategoryDraft;
+import io.sphere.sdk.taxcategories.TaxCategoryDraftBuilder;
+import io.sphere.sdk.taxcategories.commands.updateactions.SetKey;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class TaxCategorySyncOptionsTest {
+
+    private static SphereClient CTP_CLIENT = mock(SphereClient.class);
+
+    @Test
+    void applyBeforeUpdateCallback_WithNullCallback_ShouldReturnIdenticalList() {
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .build();
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNull();
```

Comment:
IMO, this assertion should not be there.

- If it fails, then it is due to another reason than what this actual unit test is testing.
- It adds a bit confusion to what this test should actually be asserting.
- It is kind of redundant in this case.

Reply:
It is just to double-check if callback is set or not, but you are right - it is confusing. I will remove this here and in other places where it is not really needed.